lib/bloc/account/account_bloc.dart

```diff
@@ -231,6 +238,68 @@ class AccountBloc {
     action.resolve(await _breezLib.sendWalletCoins(action.destAddress, Int64(action.feeRate)));     
   }
 
+  Future _exportPaymentsAction(ExportPayments action) async {
+    action.resolve(await _exportPayments());
+  }
+
+  Future _exportPayments() async {
+    // convert payment list to csv format and save to disk
+    return await _saveCsvFile(const ListToCsvConverter().convert(_generatePaymentList()));
+  }
+
+  _generatePaymentList(){
+    List<List<dynamic>> paymentList = new List.generate(_filterPayments(_paymentsController.value.paymentsList).length, (index) {
+      List paymentItem = new List();
+      paymentItem.add(DateUtils.formatYearMonthDayHourMinute(DateTime.fromMillisecondsSinceEpoch(
+          _filterPayments(_paymentsController.value.paymentsList).elementAt(index).creationTimestamp.toInt() * 1000)));
```

Comment:
No reason for calling `_filterPayments` for every item in the list...

Reply:
[👍](https://github.com/breez/breezmobile/pull/203/commits/bddf37b4321b683346c668685f19908672cf3537)